gemrb/core/Map.cpp

```diff
@@ -2503,27 +2503,26 @@ bool Map::SpawnsAlive() const
 
 void Map::PlayAreaSong(int SongType, bool restart, bool hard) const
 {
-	//Ok, we use a non constant pointer here, so it is easy to disable
-	//a faulty music list on the fly. I don't want to add a method just for that
-	//crap when we already have that pointer at hand!
-	char* poi = core->GetMusicPlaylist(SongList[SongType]);
+	size_t pl = SongList[SongType];
+	const ieVariable* poi = &core->GetMusicPlaylist(pl);
 	// for subareas fall back to the main list
 	// needed eg. in bg1 ar2607 (intro candlekeep ambush south)
 	// it's not the correct music, perhaps it needs the one from the master area
 	// it would match for ar2607 and ar2600, but very annoying (see GetMasterArea)
 	// ... but this is also definitely wrong for iwd
-	if (!poi && !MasterArea && SongType == SONG_BATTLE && core->HasFeature(GF_BREAKABLE_WEAPONS)) {
-		poi = core->GetMusicPlaylist(SongType);
+	if (!MasterArea && SongType == SONG_BATTLE && core->HasFeature(GF_BREAKABLE_WEAPONS)) {
```

Comment:
This is missing the previous !poi guard and will cause a bunch of wrong music to be played.

Reply:
its impossible to be `nullptr` tho. guess needs to become `IsStar(*poi)`